jest resolver resolves to wrong file

### Current Behavior

When resolving path './foo.js', if './foo.js' does not exist but '../foo.js' exists, the resolver will incorrectly resolve to '../foo.js'.

### Expected Behavior

resolver should be unable to resolve the file

### GitHub Repo

https://github.com/nrwl/nx

### Steps to Reproduce

1. Use`npx create-nx-workspace@latest` to create an angular standalone app called `foo`. Accept all defaults.
2. In `src/app/app.component.spec.ts`, add a mock as follows
    `jest.mock('./main', () => undefined);`
3. Try `npx nx test foo`

**Expected result**: test suite should fail to run, with an error `Cannot find module './main' from 'src/app/app.component.spec.ts'`

**Actual result**: The tests will pass. Tracing the code, the file found by '@nrwl/jest/plugins/resolver' is the existing file `src/main.ts`, i.e. `../main.ts`.


### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.14.0
   OS   : darwin x64
   npm  : 8.3.1
   
   nx                      : 15.7.2
   @nrwl/jest              : 15.7.2
   @nrwl/linter            : 15.7.2
   @nrwl/workspace         : 15.7.2
   @nrwl/angular           : 15.7.2
   @nrwl/cli               : 15.7.2
   @nrwl/cypress           : 15.7.2
   @nrwl/devkit            : 15.7.2
   @nrwl/eslint-plugin-nx  : 15.7.2
   @nrwl/js                : 15.7.2
   @nrwl/tao               : 15.7.2
   @nrwl/webpack           : 15.7.2
   @nrwl/nx-cloud          : 15.1.1
   typescript              : 4.8.4
```


### Failure Logs

_No response_

### Additional Information

This line in the jest resolver seems suspicious to me

        return ts.resolveModuleName(path, options.basedir, compilerOptions, host)
            .resolvedModule.resolvedFileName;

`options.basedir` has the value `src/app` which is a directory, but the receiving parameter in `ts.resolveModuleName` is named `containingFile`. If `src/app` is considered as a file, then `src/main.ts` would be a good resolution for `./main.ts`.

Changing this line to

        return ts.resolveModuleName(path, options.basedir + '/some-file', compilerOptions, host)
            .resolvedModule.resolvedFileName;

gives the expected behaviour, `./main' cannot be resolved anymore.

--- a/packages/jest/plugins/resolver.ts
+++ b/packages/jest/plugins/resolver.ts
@@ -1,4 +1,4 @@
-import { dirname, extname } from 'path';
+import { dirname, extname, join } from 'path';
 import { resolve as resolveExports } from 'resolve.exports';
 import type defaultResolver from 'jest-resolve/build/defaultResolver';
 
@@ -81,7 +81,11 @@ module.exports = function (path: string, options: ResolveOptions) {
     ts = ts || require('typescript');
     compilerSetup = compilerSetup || getCompilerSetup(options.rootDir);
     const { compilerOptions, host } = compilerSetup;
-    return ts.resolveModuleName(path, options.basedir, compilerOptions, host)
-      .resolvedModule.resolvedFileName;
+    return ts.resolveModuleName(
+      path,
+      join(options.basedir, 'fake-placeholder.ts'),
+      compilerOptions,
+      host
+    ).resolvedModule.resolvedFileName;
   }
 };

--- a/scripts/patched-jest-resolver.js
+++ b/scripts/patched-jest-resolver.js
@@ -91,7 +91,11 @@ module.exports = function (path, options) {
     } else if (path === '@nrwl/devkit/testing') {
       return join(__dirname, '../', './packages/devkit/testing.js');
     }
-    return ts.resolveModuleName(path, options.basedir, compilerOptions, host)
-      .resolvedModule.resolvedFileName;
+    return ts.resolveModuleName(
+      path,
+      join(options.basedir, 'fake-placeholder.ts'),
+      compilerOptions,
+      host
+    ).resolvedModule.resolvedFileName;
   }
 };

